chainerx_cc/chainerx/native/native_device/memory.cc

```diff
@@ -15,7 +15,11 @@ std::shared_ptr<void> NativeDevice::Allocate(size_t bytesize) {
     if (bytesize == 0) {
         return std::shared_ptr<void>{nullptr};
     }
+#ifdef __APPLE__
+    return std::shared_ptr<uint8_t>(new uint8_t[bytesize], std::default_delete<uint8_t[]>());
```

Comment:
Actually, you can skip the branching and simply replace the existing `std::make_unique` with the one you're proposing.

Reply:
To be insanely picky `std::make_unique` will allocate the memory regions for the `uint8_t*` and `std::unique_ptr` simultaneously so it should in theory provide a slight performance edge (I'm not sure as to exactly how much of an impact this will have in real-life situations).

Reply:
Ah yes but the return type of this function is a `std::shared_ptr<void>` so there're still addition allocations.

Reply:
Oh yes that is true I don't know why that didn't occur to me...